arviz/tests/test_data.py

```diff
@@ -176,12 +177,9 @@ def test_addition():
     idata2 = from_dict(prior={"C": np.random.randn(2, 10, 2), "D": np.random.randn(2, 10, 5, 2)})
     new_idata = idata1 + idata2
     assert new_idata is not None
-    assert hasattr(new_idata, "posterior")
-    assert hasattr(new_idata, "prior")
-    assert hasattr(new_idata.posterior, "A")
-    assert hasattr(new_idata.posterior, "B")
-    assert hasattr(new_idata.prior, "C")
-    assert hasattr(new_idata.prior, "D")
+    test_dict = {"posterior": ["A", "B"], "prior": ["C", "D"]}
+    fails = check_multiple_attrs(test_dict, new_idata)
+    assert len(fails) == 0

```

Comment:
I would change this line to  `assert not fails` for two reasons. One is that the code matches the english expectation in that we don't want any fails. The second is that when pytest produces an assertion failure we can actually see the failed attrs. Right now if it fails there we'd only see something like 
> assertationerror: 7 != 2

Reply:
Okey, no problem. As I explained above, I prioritized seeing how many checks failed because none of the approaches showed which ones failed with the old `check_multiple_attrs` because the list was too long and therefore not fully printed, now all the approaches will show all the failed checks tough, either on the first or the second line of the message.
If you agree I will start to replace all multiple checks with this function and `assert not fails`.

Reply:
Feel free to start replacing all the asserts.

The length of the diff in pytest is configurable, but either way since all the comparison logic is contained in one function, if a logic change is needed it only needs to be changed in one place! Great design :)


https://docs.pytest.org/en/latest/assert.html#making-use-of-context-sensitive-comparisons